proxy.c

```diff
@@ -0,0 +1,155 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/socket.h>
+#include <pthread.h>
+#include <limits.h>
+#include <netdb.h>
+
+#define PORT 80
+#define BUFF_SIZE 8196
+#define THREADS_COUNT 10
+
+pthread_cond_t* cv;
+pthread_mutex_t *mutex;
+int threads;
+
+
+void FoundHeaderHost(char* request, char* host){
+	char* header_host;
+	header_host = strstr(request, "Host: ");
+    if (!header_host){
+        printf("Host: header not found\n");
+        return;
+    }
+	
+	char* header_host_end = strstr(header_host, "\r\n");
+	int len = header_host_end - header_host - 6;
+	strncpy(host, header_host + 6, len);
+	host[len] = '\0';
+}
+
+int ConnectToServer(char* host) {
+    int sock_server;
+	struct sockaddr_in addr;
+
+	struct hostent* host_info = gethostbyname(host);
+	sock_server = socket(AF_INET, SOCK_STREAM, 0);
+    addr.sin_family = AF_INET;
+    addr.sin_port = htons(PORT);
+    addr.sin_addr = *((struct in_addr *)host_info->h_addr);
+
+    if (connect(sock_server, (struct sockaddr*)&addr, sizeof(addr)) < 0) {
+        perror("Connect to server error");
+        return 0;
+    }
+	
+    return sock_server;
+}
+
+void* ClientFunc(void* socket_client) {
+    int sock_client = *(int*)socket_client;
+    char buff[BUFF_SIZE + 1];
+    int sock_server;
+
+	//первый запрос с хостом сервера
+	int size = 0;
+    char host[HOST_NAME_MAX + 1];
+	while (1) {
+        bytes = recv(client_sock, buff, BUFF_SIZE, 0);
+        if (bytes <= 0) {
+			break;
+        }
+
+        if (strstr(buff, "\r\n\r\n") != NULL && !sock_server) {
```

Comment:
and now you can meet situation when first chiunk contain first \r\n and second chunk of data read with recv  contain second \r\n

Reply:
fixed